ReplayMessenger.device.lib.nut

```diff
@@ -93,6 +109,41 @@ class ReplayMessenger {
         return _mm.send(messageName, data, null, _retryInterval, metadata);
     }
 
+    function onReplayBegin(loggerName, callback = null) {
+      if (callback == null && typeof loggerName == "string") { //de-register callback
+        if (loggerName in _onReplayBegin) {
+          delete _onReplayBegin[loggerName]
+        }
+      } else if (callback == null) { //assume callback is provided for "default" logger
+        callback = loggerName;
+        loggerName = "default";
+      } else {
+        _onReplayBegin[loggerName] <- callback;
+      }
+    }
+
+    function onReplayComplete(loggerName, callback = null) {
+      if (callback == null && typeof loggerName == "string") { //de-register callback
+        if (loggerName in _onReplayComplete) {
+          delete _onReplayComplete[loggerName]
+        }
+      } else if (callback == null) { //assume callback is provided for "default" logger
```

Comment:
The same as above (copy paste?).

I'd also suggest to get rid of this code duplication as we already got into an issue because of that.